client/src/components/_common/Section/SectionMessages.js

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  WelcomeMessage,
+  shouldShowMessage as shouldShowWelcomeMessage
+} from '_common';
+import * as MESSAGES from '../../../constants/welcomeMessages';
+
+import './SectionMessages.module.css';
+import './SectionMessages.css';
+
+/**
+ * A list for section messages that supports:
+ *
+ * - manual messages
+ * - manual welcome message
+ * - automatic welcome message
+ * - automatic welcome message with custom text
+ *
+ * @example
+ * // an automatic welcome message (if found), no additional messages
+ * <SectionMessages routeName="DASHBOARD" />
+ * @example
+ * // overwrite text of an automatic welcome message, no additional messages
+ * <SectionMessages
+ *   routeName="DASHBOARD"
+ *   welcomeText={`We welcome you to the dashboard, ${givenName}`} />
+ * @example
+ * // define text for a manual welcome message, no additional messages
+ * <SectionMessages welcomeText={`We welcome you to this page, ${givenName}`} />
+ * @example
+ * // an automatic welcome message (if found), some additional messages
+ * <SectionMessages routeName="DASHBOARD">
+ *   <Alert color="success">You win!</Alert>
+ *   <Alert color="secondary">
+ *     <button>Claim your prize.</button>
+ *   </Alert>
+ * </SectionMessages>
+ * @example
+ * // no automatic welcome message, some additional messages
+ * <SectionMessages>
+ *   <Alert color="success">You win!</Alert>
+ *   <Alert color="secondary">
+ *     <button>Claim your prize.</button>
+ *   </Alert>
+ * </SectionMessages>
+ */
+function SectionMessages({ children, className, routeName, welcomeText }) {
+  const welcomeMessageText = welcomeText || MESSAGES[routeName];
+  /* FAQ: An alternate message name allows tracking custom message dismissal */
+  const welcomeMessageName = routeName || welcomeMessageText;
+  const welcomeMessage = welcomeMessageText && (
+    <WelcomeMessage messageName={welcomeMessageName}>
+      {welcomeMessageText}
+    </WelcomeMessage>
+  );
+  const hasMessage = shouldShowWelcomeMessage(routeName) || children.length > 0;
+  const hasMessageClass = 'has-message';
+
+  useEffect(() => {
+    if (hasMessage) {
+      document.body.classList.add(hasMessageClass);
+    } else {
+      document.body.classList.remove(hasMessageClass);
+    }
+  }, [hasMessage]);
+
+  return (
+    <aside styleName="root" className={className}>
```

Comment:
TIL `<aside>` is an element

Reply:
Yay!

It is an interesting HTML5 addition I am always eager to use, but [the intended use case](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/aside) ([usage note](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/aside#using_%3Caside%3E)) is rare in applications. I think I finally found one.